airgun/entities/redhat_repository.py

```diff
@@ -29,6 +29,14 @@ def read(self, entity_name, category='Available', widget_names=None):
             entity_data['items'] = view.available.items(name=entity_name)[0].read_items()
         return entity_data
 
+    def recommended_repos(self):
+        """Click on 'Recommended Repositories'
+        and read RH Repositories values.
+        """
+        view = self.navigate_to(self, 'All')
+        view.recommended_repos.click()
+        return view.available.read()
```

Comment:
![image](https://user-images.githubusercontent.com/3190629/65144852-d4cf7b80-da35-11e9-8374-06c66aac5854.png)

After successful iteration of the first attempt of reading recommended repos, Consider case here that, if someone wants to read this second time. On the second time, it will click on view.recommended_repos.click() unknowingly recognising it is in recommend the view. 

you can '$x("//div[contains(@class, 'bootstrap-switch wrapper')]")' locator get the current status e.g.

```suggestion
        view.recommended_repos.click()
        return view.available.read()
```
     if 'off' in view.recommended_repos.get_attribute("class")
          view.recommended_repos.click()
     return view.available.read()
.     

```

Reply:
Also, what if I try to call `read()` after calling this new method? If repo that I try to read is not recommended, method will fail to provide information, as knob is turned on.

I think that this new method should be more generic. Instead of reading recommended repos, it should focus on reading repos list. Additional parameters would control if it should read Available repos, Enabled repos, or both. In case of Available, another parameter should control if "Recommended repos" should be turned on or off. As far as I can tell, `RedHatRepositoriesView` already has infrastructure for reading all items in the view (through `RepositoryCategoryView.items()`), but we always filter that list in some way.

Moreover, `read()` method should be extended to check state of "Recommended repos" and turn it off in case it is turned on (or just set it to "off" and let widget figure out if it has to do anything).

Reply:
Thank you @omkarkhatavkar and @mirzal ...
I have refactor it by using your ideas and some mine too :) and now it will as below : 

- It would be part of `.read()` now.
- By-default behaviour of `.read()` will remain same.
- `recommended_repo` will work if specified specifically i.e.
```
(Pdb) on_rrepos = session.redhatrepository.read(recommended_repo='on')
(Pdb) off_rrepos = session.redhatrepository.read(recommended_repo='off')
```
- As `Recommended  Repositories` only works for `available` repositories hence worth to retrun the `list of available repos` instead of `enabled` or something else.

Reply:
@vijay8451 new logic looks okay to me.